mintel/kube-monitoring-mixins
# Initial Work on Alerts for paging future

https://gitlab.com/mintel/satoshi/planning/issues/721

This is a big rework and changelog is highligted here 
* rework the gke-override to use external functions rather then inline
* add function to gke-override to change `FOR` fields in alerts
* add function to gke-override to `DOWNGRADE` severity for an alert
* add function to gke-override to set `page:false` label, critical with this set to false will not page in opsgenie ( to be configured there ) , a value of true or a lack of label will page 
* Increase threshold for cpu throttling to 60 ( was 25 ) and re-enable alerting for it ( to be done in alertmanager ) 
* Ignore `ElasticsearchTooFewNodesRunning` and `ElasticsearchTooFewMastersRunning` alerts - We have other alerts ( like cluster status and cluster health )
* Ignore group `kube-prometheus-node-alerting.rules` , it only has duplicate rules we have in other groups 
* set all the following alerts to `page:false`
  *  'AlertmanagerConfigInconsistent',
  *  'ContainerCombinedIoHighOverTime',
  *  'ClusterLowMemoryAvailable',
  *  'ElasticsearchClusterHealthYELLOW',
  *  'ElasticsearchHeapTooHigh',
  *  'ElasticsearchLowDiskFree',
  *  'HAProxyFrontendSessionUsage',
  *  'HAProxyBackendResponseErrors',
  *  'HAProxyBackendDown',
  *  'HAProxyServerInBackendUpPercentageLow',
  *  'KubePersistentVolumeInodeUsageCritical',
  *  'KubePersistentVolumeFullInFourDays',
  *  'PodOOMKilled',
  *  'PrometheusBadConfig',
  *  'PrometheusErrorSendingAlertsToAnyAlertmanager',
  *  'PrometheusRuleFailures',
  *  'PrometheusRemoteStorageFailures',
  *  'PrometheusRemoteWriteBehind',
* Increase FOR in for our rules  ( see the diff for the list ) 
* create record for `haproxy:http_frontend_session_usage:percentage` 
  * USe in our current rules
  * do not page on Session usage only for haproxy, use a prection to fill up in 2 h. see HAProxyFrontendSessionUsageWillRunOut
* update list of excluded frontends in haproxy rules
* increase threshold for haproxy beckend queue rules
* add new alerrt `HAProxyServerInBackendUpPercentageLow` to detect a low number of server up in the pool ( not paging )
* add new alert `HAProxyFrontendIncreasedRequests` to do anomaly detection on number of requests. This is a test and is warning. 
* Update mintel PVC rules to 
  * use recorded rules and changed threshold
  * update prediction to only alert if the current available is lower then threshold and will fill up in 4 hours 
* update MintelWebServiceDown and MintelWebServiceReduced to actually work ( i hope )
*
